app/src/androidTest/java/ch/epfl/sdp/login/RegisterTest.java

```diff
@@ -135,47 +133,33 @@ public void registering_ShouldFailOnEmptyTextFields() {
                 onView(withId(emptyFields.get(i))).check(matches(hasErrorText(errorTexts.get(i))));
             }
             Log.d("COUNTER", " " + i);
-            onView(withId(R.id.backBtn)).perform(click());
-            onView(withId(R.id.createAccountBtn)).perform(click());
+            onView(withId(R.id.registerbutton)).perform(click());
             onView(withId(R.id.email)).check(matches(isDisplayed()));
+
+            onView(withId(R.id.username)).perform(clearText());
+            onView(withId(R.id.txtRegisterPassword)).perform(clearText());
+            onView(withId(R.id.passwordconf)).perform(clearText());
+            onView(withId(R.id.email)).perform(clearText());
         }
     }
 
     // for now
     @Test
     public void registering_ShouldFailOnPasswordSmallerThan8() {
-        MissingFieldTestFactory.testFieldFourActions(new Pair(typeText("a"), R.id.username), new Pair(typeText("a@a"), R.id.email), new Pair(typeText("passwor"), R.id.password), new Pair(click(), R.id.passwordconf));
+        MissingFieldTestFactory.testFieldFourActions(new Pair(typeText("a"), R.id.username), new Pair(typeText("a@a"), R.id.email), new Pair(typeText("passwor"), R.id.txtRegisterPassword), new Pair(click(), R.id.passwordconf));
         closeSoftKeyboard();
         onView(withId(R.id.registerbutton)).perform(click());
-        onView(withId(R.id.password)).check(matches(hasErrorText("Password is incorrect")));
+        onView(withId(R.id.txtRegisterPassword)).check(matches(hasErrorText("Password is incorrect")));
     }
 
     @Test
     public void registering_ShouldWorkOnNewCorrectInformation() {
         String newUsername = "Username";
         String newEmail = "Email@a";
-        MissingFieldTestFactory.testFieldFourActions(new Pair(typeText(newUsername), R.id.username), new Pair(typeText(newEmail), R.id.email), new Pair(typeText(password), R.id.password), new Pair(typeText(password), R.id.passwordconf));
+        MissingFieldTestFactory.testFieldFourActions(new Pair(typeText(newUsername), R.id.username), new Pair(typeText(newEmail), R.id.email), new Pair(typeText(password), R.id.txtRegisterPassword), new Pair(typeText(password), R.id.passwordconf));
         closeSoftKeyboard();
         intending(toPackage(MainActivity.class.getName())).respondWith(result);
         onView(withId(R.id.registerbutton)).perform(click());
         onView(withId(R.id.rulesButton)).check(matches(isDisplayed()));
     }
-
-    // for now
-
-    @Test
-    public void backButton_ShouldGoToLoginForm() {
```

Comment:
I think I kept some tests here in my branch

Reply:
I delete the back button tests because it doesnt exist. There were another test which was uncommented. That one I also deleted.